src/internal/observable/dom/fetch.ts

```diff
@@ -0,0 +1,61 @@
+import { Observable } from '../../Observable';
+
+/**
+ * Uses [the Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) to
+ * make an HTTP request.
+ *
+ * Will automatically set up an internal [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController)
+ * in order to teardown the internal `fetch` when the subscription tears down.
+ *
+ * If a `signal` is provided via the `init` argument, it will behave like it usually does with
+ * `fetch`. If the provided `signal` aborts, the error that `fetch` normally rejects with
+ * in that scenario will be emitted as an error from the observable.
+ *
+ * @param input The resource you would like to fetch. Can be a url or a request object.
+ * @param init A configuration object for the fetch.
+ * [See MDN for more details](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters)
+ * @returns An Observable, that when subscribed to performs an HTTP request using the native `fetch`
+ * function. The {@link Subscription} is tied to an `AbortController` for the the fetch.
+ */
+export function fromFetch(input: string | Request, init?: RequestInit): Observable<Response> {
+  return new Observable<Response>(subscriber => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+    let outerSignalHandler: () => void;
+    let unsubscribed = false;
+
+    if (init) {
+      // If we a signal is provided, just have it teardown. It's a cancellation token, basically.
+      if (init.signal) {
+        outerSignalHandler = () => {
+          if (!signal.aborted) {
+            controller.abort();
+          }
+        };
+        init.signal.addEventListener('abort', outerSignalHandler);
+      }
+      init.signal = signal;
+    } else {
+      init = { signal };
+    }
+
+    fetch(input, init).then(response => {
+      if (response.ok) {
+        subscriber.next(response);
+        subscriber.complete();
+      } else {
+        subscriber.error(new Error(`RxJS Fetch HTTP Error\n\nStatus:\n${response.status}\n\nBody:\n${response.body}\n`));
```

Comment:
IMO, it would be a mistake to error on non-OK responses. This is not what `fetch` does; `fetch` only rejects if a network error occurs.

I guess the question is: should this align with `fetch`'s non-OK response handling or should it be like Angular and turn non-OK responses into errors.

I'm more inclined to go with the former, but I can see why others might favour the latter. However, if it is going to use the latter approach, IMO, it *must* make the error response available via the error.

Reply:
Thinking about this more, I think this should mimic `fetch` as much as possible. If that is going to be the defacto standard going forward.